drivers/i2c/Kconfig.esp32

```diff
@@ -0,0 +1,102 @@
+# Kconfig.esp32 - ESP32 I2C configuration options
+#
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config I2C_ESP32
+	bool "ESP32 I2C"
+	depends on I2C && SOC_ESP32 && GPIO_ESP32
+	default n
+	help
+	  Enables the ESP32 I2C driver
+
+if I2C_ESP32
+
+config I2C_ESP32_TIMEOUT
+	int "I2C timeout to receive a data bit in APB clock cycles"
+	default 200000
+
+if I2C_0
+
+config I2C_0_DEFAULT_CFG
```

Comment:
Seems obscure.  I guess this is a bitmask of a config register?  Can't you split out the fields you want configurable and compute the bitmask in a compile-time C expression?  Needs docs anyway.  No one will know what to do with this at the kconfig level.

Reply:
This is just how our I2C subsystem is configured.  I don't like this as well, and would prefer separate settings, but it's something to be fixed in another pull request.

Reply:
Got it.  Yeah, your'e right.  Not your problem to fix in this patch.

Reply:
This issue has already been fixed #1081. Drivers which use DTS can define start-up configuration in a clean way.